front-end/src/components/modal/FileStorage.js

```diff
@@ -102,21 +101,21 @@ const CustomHeaderRenderer = (props) => {
 }
 
 
-const FileStorage = ({ selectedFiles, setSelectedFiles }) => {
+const FileStorage = ({ files, selectedFiles, setSelectedFiles}) => {
     // Files is of format [{key: "name", fileHeaders: [header1, header2], size: 1234}, ...}]
     // Here is the implementation of the file browser with props passed in
     
-    // holds all the files which have been uploaded
-    const [files, setFiles] = useState([])
-
-    useEffect(() => {
-        // Fetch data when the component mounts
-        fetch(`http://${window.location.hostname}:8080/files`)
-          .then((response) => response.json())
-          .then((data) => {
-            setFiles(data.files);
-          });
-      }, []); // Empty dependency array ensures that the fetch is only performed once
+    // // holds all the files which have been uploaded
+    // const [files, setFiles] = useState([])
+
+    // useEffect(() => {
+    //     // Fetch data when the component mounts
+    //     fetch(`http://${window.location.hostname}:8080/files/folder/csv`)
+    //       .then((response) => response.json())
+    //       .then((data) => {
+    //         setFiles(data.files);
+    //       });
+    //   }, []); // Empty dependency array ensures that the fetch is only performed once
```

Comment:
My understanding here is that we fetch the data when the createGraphModal mounts, so that we don't need to wait for it anymore, right?

Reply:
Yes and this was required so that I can pass in the filtered files since I needed file storage to be able to take in an arbitrary set of files to select from